vector-stores/spring-ai-coherence-store/src/test/java/org/springframework/ai/vectorstore/CoherenceVectorStoreIT.java

```diff
@@ -0,0 +1,319 @@
+package org.springframework.ai.vectorstore;
+
+import com.oracle.bedrock.junit.CoherenceClusterExtension;
+import com.oracle.bedrock.runtime.coherence.CoherenceClusterMember;
+import com.oracle.bedrock.runtime.coherence.options.ClusterName;
+import com.oracle.bedrock.runtime.coherence.options.LocalHost;
+import com.oracle.bedrock.runtime.coherence.options.RoleName;
+import com.oracle.bedrock.runtime.coherence.options.WellKnownAddress;
+import com.oracle.bedrock.runtime.java.options.IPv4Preferred;
+import com.oracle.bedrock.runtime.java.options.SystemProperty;
+import com.oracle.bedrock.runtime.options.DisplayName;
+import com.oracle.bedrock.testsupport.junit.TestLogsExtension;
+
+import com.tangosol.net.Coherence;
+import com.tangosol.net.Session;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.*;
+
+import org.springframework.ai.document.Document;
+import org.springframework.ai.embedding.EmbeddingModel;
+import org.springframework.ai.transformers.TransformersEmbeddingModel;
+import org.springframework.ai.vectorstore.filter.FilterExpressionTextParser;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.util.CollectionUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CoherenceVectorStoreIT {
```

Comment:
Hi, I see Oracle Coherence CE has some docker images available [here](https://github.com/oracle/coherence/pkgs/container/coherence-ce).

I wonder if we can switch and use it along with Testcontainers. Later, when autoconfiguration is added then adding Service Connections support for  [Testcontainers](https://docs.spring.io/spring-ai/reference/api/testcontainers.html) and [Docker Compose](https://docs.spring.io/spring-ai/reference/api/docker-compose.html) would be great.

Reply:
We do have Docker images, but no OOB support for Testcontainers that I'm aware off, so someone would need to write that.

The thing is, we had a way to start a multi-process cluster long before Testcontainers existed, in the form of [Oracle Bedrock](https://github.com/coherence-community/oracle-bedrock), so that's what we used to test this integration as well.

I'm not saying it's better than Testcontainers, but it's lighter-weight and faster to run, as it doesn't require or has to download Docker image on every build -- it simply starts several Coherence cluster members, and runs the tests.

Anyway, I'd like to get this in as-is, so we have the integration people can use, but am open to refactoring tests to use Testcontainers long-term, if there are benefits to doing that.